File: VB-Examples/02_FindAndReplace/FindAndHighlight.vb
Imports Spire.Doc
Imports Spire.Doc.Documents

Namespace FindAndHighlight
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Create a new Document object
			Dim document As New Document()

			' Load a Word document from the specified file path
			document.LoadFromFile("..\..\..\..\..\..\Data\Sample.docx")

			' Find all occurrences of the string "word" in the document (case-insensitive)
			Dim textSelections() As TextSelection = document.FindAllString("word", False, True)

			' Iterate through each TextSelection
			For Each selection As TextSelection In textSelections
				' Set the highlight color of the selected text to yellow
				selection.GetAsOneRange().CharacterFormat.HighlightColor = Color.Yellow
			Next selection

			' Save the modified document to a file named "Sample.docx"
			document.SaveToFile("Sample.docx", FileFormat.Docx)

			' Dispose of the Document object to release resources
			document.Dispose()

			'Launching the  Word file.
			WordDocViewer("Sample.docx")
		End Sub

		Private Sub WordDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
	End Class
End Namespace

File: CS-Examples/02_FindAndReplace/FindAndHighlight.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using Spire.Doc;
using Spire.Doc.Documents;

namespace FindAndHighlight
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Create a new Document object
            Document document = new Document();

            // Load an existing Word document from the specified file path
            document.LoadFromFile(@"..\..\..\..\..\..\Data\Sample.docx");

            // Find all occurrences of the string "word" in the document and retrieve their TextSelections
            TextSelection[] textSelections = document.FindAllString("word", false, true);

            // Iterate through each TextSelection
            foreach (TextSelection selection in textSelections)
            {
                // Get the entire range of the selection and set its CharacterFormat's HighlightColor property to Yellow
                selection.GetAsOneRange().CharacterFormat.HighlightColor = Color.Yellow;
            }

            // Save the modified document to a file named "Sample.docx" in Docx format
            document.SaveToFile("Sample.docx", FileFormat.Docx);

            // Dispose of the document object and release any associated resources
            document.Dispose();

            //Launching the  Word file.
            WordDocViewer("Sample.docx");
        }

        private void WordDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}
